File: ios/Classes/resources/Repository.swift
enum RepositoryError: Error {
    case ItemNotFound
}

class Repository<T> {
    var items: [String: T] = [:]
    
    public func get(id: String) throws -> T {
        if !exist(id: id) {
            throw RepositoryError.ItemNotFound
        }
        return items[id]!
    }
    
    public func set(id: String, item: T) {
        items[id] = item
    }
    
    private func exist(id: String) -> Bool {
        return items.keys.contains(id)
    }
    
    open func close(id: String) {
        items.removeValue(forKey: id)
    }
}

File: packages/pdfx/android/src/main/kotlin/io/scer/pdfx/resources/Repository.kt
package io.scer.pdfx.resources

import java.lang.Exception

class RepositoryItemNotFoundException(message: String): Exception(message)

abstract class Repository<T> {
    private val items: MutableMap<String, T> = HashMap()

    @Throws(RepositoryItemNotFoundException::class)
    fun get(id: String): T {
        if (!exist(id)) throw RepositoryItemNotFoundException(id)
        return items[id]!!
    }

    fun set(id: String, item: T) {
        items[id] = item
    }

    fun clear() = items.clear()

    private fun exist(id: String): Boolean = items.contains(id)

    protected open fun close(id: String) {
        items.remove(id)
    }
}
